device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+FW_UPGRADE_SCRIPT="/usr/bin/mlnx-fw-upgrade.sh"
+
+NEXT_SONIC_IMAGE="$(sonic_installer list | grep "Next: " | cut -d ' ' -f 2)"
+CURRENT_SONIC_IMAGE="$(sonic_installer list | grep "Current: " | cut -d ' ' -f 2)"
+
+FORCE_REBOOT="no"
+
+ParseArguments() {
+    while [ $# -ge 1 ]; do
+        case "$1" in
+            -f|--force)
+                FORCE_REBOOT="yes"
+            ;;
+        esac
+        shift
+    done
+}
+
+ParseArguments $@
+
+if [[ "${CURRENT_SONIC_IMAGE}" != "${NEXT_SONIC_IMAGE}" ]]; then
```

Comment:
Can you reuse code in https://github.com/Azure/sonic-utilities/blob/8819b4f7445f6389c3ceea1ee667993f2b586eba/scripts/fast-reboot#L39-L58